### PR TITLE
add new property pane button and zoom button

### PR DESCRIPTION
**Pull request by ram-primathon**
## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/components/designSystems/appsmith/StepComponent.tsx [hunk 1]
```diff
@@ -2,6 +2,8 @@ import React from "react";
 import { ControlIcons } from "icons/ControlIcons";
```
[the remark this answers is not in the document]
Comment (hetunandu) [reply app/client/src/components/designSystems/appsmith/StepComponent.tsx:2]: Revert all changes in this file

File: app/client/src/components/ads/StepComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  width: 60px;
+  height: 32px;
+`;
+
+const StepWrapper = styled.div<{ skin: Skin }>`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  background: ${(props) => props.theme.ads.stepComponent[props.skin].stepBG};
```
Comment (hetunandu) [inline app/client/src/components/ads/StepComponent.tsx:35]: Fix the place where this is imported from

File: app/client/src/components/propertyControls/PrimaryColumnsControl.tsx [hunk 3]
```diff
@@ -201,8 +201,8 @@ class PrimaryColumnsControl extends BaseControl<ControlProps> {
           className="t--add-column-btn"
           text="Add a new column"
           icon="plus"
-          color="#FFFFFF"
-          minimal={true}
+          // color="#FFFFFF"
```
Comment (hetunandu) [inline app/client/src/components/propertyControls/PrimaryColumnsControl.tsx:204]: Remove these lines

File: app/client/src/components/propertyControls/StepControl.tsx [hunk 4]
```diff
@@ -26,6 +27,8 @@ class StepControl extends BaseControl<StepControlProps> {
   };
 
   render() {
+    console.log(this.props);
```
Comment (hetunandu) [inline app/client/src/components/propertyControls/StepControl.tsx:30]: remove console.log

File: app/client/src/components/propertyControls/StepControl.tsx [hunk 5]
```diff
@@ -37,6 +40,7 @@ class StepControl extends BaseControl<StepControlProps> {
           this.updateProperty(this.props.propertyName, value);
         }}
         displayFormat={displayFormat}
+        skin={this.props.theme === EditorTheme.DARK ? Skin.DARK : Skin.LIGHT}
```
Comment (hetunandu) [inline app/client/src/components/propertyControls/StepControl.tsx:43]: Dont pass skin or theme

File: app/client/src/components/propertyControls/StyledControls.tsx [hunk 6]
```diff
@@ -57,16 +56,23 @@ export const ControlPropertyLabelContainer = styled.div`
...
+
+  &&& svg {
+    height: auto;
+    width: 28px;
     rect {
       fill: ${(props) =>
         props.active
-          ? props.theme.colors.primaryOld
-          : props.theme.colors.paneIcon};
+          ? props.theme.colors.jsButtonActiveColor
```
Comment (hetunandu) [inline app/client/src/components/propertyControls/StyledControls.tsx:74]: USe the correct colors

File: app/client/src/constants/DefaultTheme.tsx [hunk 7]
```diff
@@ -2049,6 +2077,32 @@ export const theme: Theme = {
...
+      [Skin.LIGHT]: {
+        colorBG: Colors.LIGHT_GREY,
+        colorText: Colors.CODE_GRAY,
+      },
+      [Skin.DARK]: {
+        colorBG: Colors.CHARCOAL,
+        colorText: Colors.WHITE,
+      },
+    },
+  },
```
Comment (hetunandu) [inline app/client/src/constants/DefaultTheme.tsx:2105]: ads does not need a new key. Add this inside light and dark const in DefaultTheme

File: app/client/src/constants/DefaultTheme.tsx [hunk 8]
```diff
@@ -1895,6 +1921,8 @@ export const theme: Theme = {
       array: "#CDFFA5",
       number: "#FFB2B2",
     },
+    jsButtonColor: Colors.SILVER_CHALICE,
```
Comment (hetunandu) [inline app/client/src/constants/DefaultTheme.tsx:1924]: Add this under the propertyPane key in light and dark DefaultTheme

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/constants/DefaultTheme.tsx (hunk 8 above)
Comment (ram-primathon) [reply app/client/src/constants/DefaultTheme.tsx]: done

File: app/client/src/constants/DefaultTheme.tsx (hunk 7 above)
Comment (ram-primathon) [reply app/client/src/constants/DefaultTheme.tsx]: done

File: app/client/src/components/propertyControls/PrimaryColumnsControl.tsx [hunk 9]
```diff
@@ -197,13 +199,14 @@ class PrimaryColumnsControl extends BaseControl<ControlProps> {
           toggleVisibility={this.toggleVisibility}
           onEdit={this.onEdit}
         />
+
         <AddColumnButton
-          className="t--add-column-btn"
```
Comment (hetunandu) [inline app/client/src/components/propertyControls/PrimaryColumnsControl.tsx:201]: Keep this class name. Classnames with `t--` prefix are used for automated testing

File: app/client/src/components/propertyControls/StyledControls.tsx [hunk 10]
```diff
@@ -57,16 +56,44 @@ export const ControlPropertyLabelContainer = styled.div`
...
-  height: 24px;
-  svg {
-    height: 24px;
+  border-radius: 4px;
+  height: auto;
+  width: 28px;
+  height: 16px;
+  border: 0.5px solid #6a86ce;
+  background-color: ${(props) =>
+    props.active ? "#6A86CE" : props.theme.colors.propertyPane.addButtonBG};
```
Comment (hetunandu) [inline app/client/src/components/propertyControls/StyledControls.tsx:67]: Add the active color as part of the theme as well

File: app/client/src/constants/DefaultTheme.tsx [hunk 11]
```diff
@@ -893,6 +893,9 @@ type ColorType = {
     radioGroupBg: ShadeColor;
     radioGroupText: ShadeColor;
     deleteIconColor: string;
+    zoomButtonBG: string;
```
Comment (hetunandu) [inline app/client/src/constants/DefaultTheme.tsx:896]: Should be ShadeColor as well.

File: app/client/src/constants/DefaultTheme.tsx [hunk 12]
```diff
@@ -1714,6 +1720,9 @@ export const light: ColorType = {
     radioGroupBg: lightShades[0],
     radioGroupText: lightShades[8],
     deleteIconColor: "#A3B3BF",
+    zoomButtonBG: "#E0DEDE",
+    addButtonBG: lightShades[11],
```
Comment (hetunandu) [inline app/client/src/constants/DefaultTheme.tsx:1724]: Use buttonText instead of addButtonBG.

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/components/propertyControls/KeyValueComponent.tsx [hunk 13]
```diff
@@ -173,12 +174,14 @@ export function KeyValueComponent(props: KeyValueComponentProps) {
           </StyledOptionControlWrapper>
         );
       })}
+
       <StyledPropertyPaneButton
+        tag="button"
+        type="button"
         text={props.addLabel || "Option"}
-        icon={"plus"}
```
Comment (hetunandu) [inline app/client/src/components/propertyControls/KeyValueComponent.tsx:178]: Add back icon. Icon is needed

File: app/client/src/components/propertyControls/PrimaryColumnsControl.tsx [hunk 14]
```diff
@@ -197,13 +199,14 @@ class PrimaryColumnsControl extends BaseControl<ControlProps> {
...
+          tag="button"
+          type="button"
           text="Add a new column"
-          icon="plus"
```
Comment (hetunandu) [inline app/client/src/components/propertyControls/PrimaryColumnsControl.tsx:203]: icon is still needed

File: app/client/src/components/propertyControls/ColumnActionSelectorControl.tsx [hunk 15]
```diff
@@ -81,12 +81,14 @@ class ColumnActionSelectorControl extends BaseControl<
               </div>
             );
           })}
+
         <StyledPropertyPaneButton
-          text={"New Button"}
-          icon={"plus"}
```
Comment (hetunandu) [inline app/client/src/components/propertyControls/ColumnActionSelectorControl.tsx:86]: icon is still needed

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/components/propertyControls/PrimaryColumnsControl.tsx (hunk 9 above)
Comment (ram-primathon) [reply app/client/src/components/propertyControls/PrimaryColumnsControl.tsx:201]: done

File: app/client/src/components/propertyControls/StyledControls.tsx (hunk 10 above)
Comment (ram-primathon) [reply app/client/src/components/propertyControls/StyledControls.tsx:67]: done

File: app/client/src/constants/DefaultTheme.tsx (hunk 12 above)
Comment (ram-primathon) [reply app/client/src/constants/DefaultTheme.tsx:1724]: done

File: app/client/src/constants/DefaultTheme.tsx (hunk 11 above)
Comment (ram-primathon) [reply app/client/src/constants/DefaultTheme.tsx:896]: nope

File: app/client/src/components/propertyControls/KeyValueComponent.tsx (hunk 13 above)
Comment (ram-primathon) [reply app/client/src/components/propertyControls/KeyValueComponent.tsx:178]: done

File: app/client/src/components/propertyControls/PrimaryColumnsControl.tsx (hunk 14 above)
Comment (ram-primathon) [reply app/client/src/components/propertyControls/PrimaryColumnsControl.tsx:203]: done

File: app/client/src/components/propertyControls/ColumnActionSelectorControl.tsx (hunk 15 above)
Comment (ram-primathon) [reply app/client/src/components/propertyControls/ColumnActionSelectorControl.tsx:86]: done

File: app/client/src/constants/DefaultTheme.tsx (hunk 11 above)
Comment (hetunandu) [reply app/client/src/constants/DefaultTheme.tsx:896]: This is still string, should be of type ShadeColor

File: app/client/src/constants/DefaultTheme.tsx [hunk 16]
```diff
@@ -893,6 +893,10 @@ type ColorType = {
     radioGroupBg: ShadeColor;
     radioGroupText: ShadeColor;
     deleteIconColor: string;
+    zoomButtonBG: string;
+    addButtonBG: ShadeColor;
+    activeButtonText: string;
```
Comment (hetunandu) [inline app/client/src/constants/DefaultTheme.tsx:898]: Should be of type ShadeColor

File: app/client/src/constants/DefaultTheme.tsx (hunk 12 above)
Comment (hetunandu) [reply app/client/src/constants/DefaultTheme.tsx:1726]: Yes, what i meant was dont add this new key at all. Use the buttonText value everywhere instead

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/constants/DefaultTheme.tsx (hunk 11 above)
Comment (hetunandu) [reply app/client/src/constants/DefaultTheme.tsx:896]: This is still string, should be of type ShadeColor

Comment (ram-primathon) [reply app/client/src/constants/DefaultTheme.tsx:896]: One color is zoomButtonBG: "#E0DEDE" which is not present in type ShadeColor then how am I suppose to keep its type as ShadeColor

File: app/client/src/constants/DefaultTheme.tsx (hunk 12 above)
Comment (ram-primathon) [reply app/client/src/constants/DefaultTheme.tsx:1726]: done

File: app/client/src/constants/DefaultTheme.tsx (hunk 11 above)
Comment (hetunandu) [reply app/client/src/constants/DefaultTheme.tsx:896]: You will need to add the light and dark theme colour in `lightShades` and `darkShades` array and use that instead. That will fix the type issue

File: app/client/src/constants/DefaultTheme.tsx (hunk 16 above)
Comment (ram-primathon) [reply app/client/src/constants/DefaultTheme.tsx]: done

File: app/client/src/constants/DefaultTheme.tsx (hunk 11 above)
Comment (ram-primathon) [reply app/client/src/constants/DefaultTheme.tsx]: done

File: app/client/src/constants/Colors.tsx [hunk 17]
```diff
@@ -72,6 +72,8 @@ export const Colors: Record<string, string> = {
   LIGHT_GREY: "#D4D4D4",
   Gallery: "#F0F0F0",
   Galliano: "#E0B30E",
+  ALTO_2: "#E0DEDE",
+  SILVER_CHALICE: "#A9A7A7",
```
Comment (hetunandu) [inline app/client/src/constants/Colors.tsx:76]: Not used anymore. Remove

File: app/client/src/components/propertyControls/StyledControls.tsx [hunk 18]
```diff
@@ -57,16 +56,50 @@ export const ControlPropertyLabelContainer = styled.div`
...
     rect {
-      fill: ${(props) =>
-        props.active
-          ? props.theme.colors.primaryOld
-          : props.theme.colors.paneIcon};
+      fill: transparent;
+    }
+
+    path {
+      fill: ${(props) => (props.active ? "#ffffff" : "#6A86CE")};
```
Comment (hetunandu) [inline app/client/src/components/propertyControls/StyledControls.tsx:102]: Use theme colour here

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/components/propertyControls/StyledControls.tsx (hunk 18 above)
Comment (ram-primathon) [reply app/client/src/components/propertyControls/StyledControls.tsx]: done

File: app/client/src/constants/Colors.tsx (hunk 17 above)
Comment (ram-primathon) [reply app/client/src/constants/Colors.tsx]: done

**Review by hetunandu — APPROVED** [review]

**Pull request merged**